### PR TITLE
vagrant changes to support fedora21

### PR DESCRIPTION
**Pull request by rajatchopra**
@danmcp Review and merge please.
Tested with multi-node setup locally.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -117,8 +117,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     # OS platform to box information
     kube_box = {
       "fedora" => {
-        "name" => "fedora20_openshift",
-        "box_url" => "https://mirror.openshift.com/pub/vagrant/boxes/openshift3/fedora_20_latest.box"
+        "name" => "fedora_vbox_deps",
```
Comment (danmcp) [inline Vagrantfile]: should name fedora_deps to match the same image from vagrant-openshift

File: Vagrantfile [hunk 2]
```diff
@@ -128,8 +128,6 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       config.vm.box_url = kube_box[kube_os]["box_url"]
       config.vm.provision "shell", inline: "/vagrant/vagrant/provision-master.sh #{master_ip} #{num_minion} #{minion_ips_str} #{ENV['OPENSHIFT_SDN']}"
       config.vm.network "private_network", ip: "#{master_ip}"
-      config.vm.hostname = "openshift-master"
```
Comment (danmcp) [inline Vagrantfile]: I'm not sure why this is being removed.

File: vagrant/provision-master.sh [hunk 3]
```diff
@@ -17,6 +19,7 @@ for (( i=0; i<${#MINION_NAMES[@]}; i++)); do
     echo "$ip $minion" >> /etc/hosts
   fi
 done
+echo "${MASTER_IP} ${MASTER_NAME}" >> /etc/hosts
```
Comment (danmcp) [inline vagrant/provision-master.sh]: Can you make reentrant?

File: Vagrantfile (hunk 2 above)
Comment (danmcp) [inline Vagrantfile]: See: https://github.com/openshift/origin/pull/2149

**Comment by rajatchopra** [comment]
@danmcp Didnt need hostname stuff after all. I do not seem to get the error on this box (used to get it on the fedora21 bento box, so I just assumed). Rebased and ready to go.

File: vagrant/provision-master.sh [hunk 4]
```diff
@@ -17,6 +17,9 @@ for (( i=0; i<${#MINION_NAMES[@]}; i++)); do
     echo "$ip $minion" >> /etc/hosts
   fi
 done
+if [ ! "$(cat /etc/hosts | grep ${MASTER_IP})" ]; then
```
Comment (danmcp) [inline vagrant/provision-master.sh]: Why cat and grep vs just grep?
and why check for "" vs checking for an exit code?

**Comment by danmcp** [comment]
[merge]

**Pull request merged**